artiq/coredevice/comm_kernel.py

```diff
@@ -628,11 +628,17 @@ def service(obj, attr, value): return setattr(obj, attr, value)
         else:
             logger.debug("rpc service: %d %r %r = %r",
                          service_id, args, kwargs, result)
-            self._write_header(Request.RPCReply)
-            self._write_bytes(return_tags)
-            self._send_rpc_value(bytearray(return_tags),
-                                 result, result, service)
-            self._flush()
+            try:
+                self._write_header(Request.RPCReply)
+                self._write_bytes(return_tags)
+                self._send_rpc_value(bytearray(return_tags),
```

Comment:
> Why is it not in the traceback anyway?

This is after the function being called by the RPC has returned so it's not on the call stack anymore.

> Can't the code that raises the error initially simply include the function information in its exception?

We could do, where ever it's an explicitly raised exception. In fact we already do in one instance, but most of the raised exceptions don't include this information. I like centralising these kind of things so that as more `raise`s get added you don't have to remember to add it each time. And this also has the advantage of adding this information to any unexpected errors. 

> I would at least narrow down the try block to only the function susceptible to fail, which is only this line AFAICT?

I agree that is the most likely function to fail. But I was feeling defensive when I wrote this and I can't see any harm in increasing the scope of this try block. The original exception is preserved as the cause of the exception raised here so we're not losing any information and the context might just be useful.